addon/src/settled.ts

```diff
@@ -190,19 +190,24 @@ if (loader.registry['ember-testing/test/waiters']) {
 function checkWaiters() {
   // eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
   type Waiter = [any, Function];
-  const EmberTest = Ember.Test as any as { waiters: Array<Waiter> };
+  const EmberTest = Ember.Test as any as {
+    waiters: Array<Waiter>;
+    checkWaiters: () => boolean;
+  };
 
   if (_internalCheckWaiters) {
     return _internalCheckWaiters();
-  } else if (EmberTest.waiters) {
+  }
+
+  if (EmberTest.waiters) {
     if (
       EmberTest.waiters.some(([context, callback]) => !callback.call(context))
     ) {
       return true;
     }
   }
 
-  return false;
+  return EmberTest.checkWaiters();
```

Comment:
changing this from `false` to `EmberTest.checkWaiters()` _is_ a meaningful change, but the only mainingful change in this PR (for consumers).

The code for `checkWaiters` is here: https://github.com/emberjs/ember.js/blob/85a4f298f67ff70395cf7f9103682335162e0606/packages/ember-testing/lib/test/waiters.ts#L111

and those callbacks are part of the legacy waiter system.

There are existing comments about removing code for accessing legacy-waiter stuff (because ember.js has since abstracted it so we don't need to be so weird in this library) -- cleaning that up can happen in a followup PR, and should be non-breaking, and retain legacy waiter support.

Reply:
So.... I suppose this should be a patch-release,  rather than non-release.